.github/workflows/main.yml

```diff
@@ -39,9 +39,23 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-node-
 
-      - name: Test
+      - name: Run unit tests
         run: ./gradlew check jibDockerBuild -x bootJar
 
+      - name: Start backend server in the background for end-to-end tests
+        run: ./gradlew bootRun &
+
+      - name: Run end-to-end tests
+        uses: cypress-io/github-action@v2
+        with:
+          working-directory: client
+          install: false
+          start: npm start
```

Comment:
We could also use the all-in-one Docker image that has just been built :thinking: 
`docker run -d cfreak/codefreak -p8080:8080 --name=cfreak-cypress`
and if something goes wrong we could print the log with `docker logs cfreak-cypress`.